NEWS.md

```diff
@@ -1,3 +1,245 @@
+  - This is a breaking change. Use `utf8bytelength/0` to get byte index.
+- Improve `tonumber/0` performance and rejects numbers with leading or trailing
+  white spaces. @itchyny @thaliaarchi #3055 #3195
+  - This is a breaking change. Use `trim/0` to remove leading and trailing white spaces.
+- Populate timezone data when formatting time. This fixes timezone name in
+  `strftime/1`, `strflocaltime/1` for DST. @marcin-serwin @sihde #3203 #3264 #3323
+- Preserve numerical precision on unary negation, `abs/0`, `length/0`. @itchyny #3242 #3275
+- Make `last(empty)` yield no output values like `first(empty)`. @itchyny #3179
+- Make `ltrimstr/1` and `rtrimstr/1` error for non-string inputs. @emanuele6 #2969
+- Make `limit/2` error for negative count. @itchyny #3181
+- Fix `mktime/0` overflow and allow fewer elements in date-time representation array. @emanuele6 #3070 #3162
+- Fix non-matched optional capture group. @wader #3238
+- Provide `strptime/1` on all systems. @george-hopkins @fdellwing  #3008 #3094
+- Fix `_WIN32` port of `strptime`. @emanuele6 #3071
+- Improve `bsearch/1` performance by implementing in C. @eloycoto #2945
+- Improve `unique/0` and `unique_by/1` performance. @itchyny @emanuele6 #3254 #3304
+- Fix error messages including long string literal not to break Unicode characters. @itchyny #3249
+- Remove `pow10/0` as it has been deprecated in glibc 2.27. Use `exp10/0` instead. @itchyny #3059
+- Remove private (and undocumented) `_nwise` filter. @itchyny #3260
+
+## Language changes
+
+- Fix precedence of binding syntax against unary and binary operators.
+  Also, allow some expressions as object values. @itchyny #3053 #3326
+  - This is a breaking change that may change the output of filters with binding syntax as follows.
+
+  ```sh
+  $ jq -nc '[-1 as $x | 1,$x]'
+  [1,-1]    # previously, [-1,-1]
+  $ jq -nc '1 | . + 2 as $x | -$x'
+  -3        # previously, -1
+  $ jq -nc '{x: 1 + 2, y: false or true, z: null // 3}'
+  {"x":3,"y":true,"z":3}    # previously, syntax error
+  ```
+
+- Support Tcl-style multiline comments. @emanuele6 #2989
+
+  ```sh
+  #!/bin/sh --
+  # Can be use to do shebang scripts.
+  # Next line will be seen as a comment be of the trailing backslash. \
+  exec jq ...
+  # this jq expression will result in [1]
+  [
+    1,
+    # \
+    2
+  ]
+  ```
+
+- Fix `foreach` not to break init backtracking with `DUPN`. @kanwren #3266
+
+  ```sh
+  $ jq -n '[1, 2] | foreach .[] as $x (0, 1; . + $x)'
+  1
+  3
+  2
+  4
+  ```
+
+- Fix `reduce`/`foreach` state variable should not be reset each iteration. @itchyny #3205
+
+  ```sh
+  $ jq -n 'reduce range(5) as $x (0; .+$x | select($x!=2))'
+  8
+  $ jq -nc '[foreach range(5) as $x (0; .+$x | select($x!=2); [$x,.])]'
+  [[0,0],[1,1],[3,4],[4,8]]
+  ```
+
+- Support CRLF line breaks in filters. @itchyny #3274
+- Improve performance of repeating strings. @itchyny #3272
+- Increase the maximum parsing depth for JSON to 10000. @itchyny #3328
```

Comment:
Grat changelog - not sure if this line should be under language changes - as I don't think JQ lang defines a max parsing depth.

Reply:
CLI changes?

Reply:
Sure, or maybe changes to existing functions?